src/main/java/ca/corefacility/bioinformatics/irida/service/impl/ProjectServiceImpl.java

```diff
@@ -265,6 +269,11 @@ public void removeUserFromProject(Project project, User user) throws ProjectWith
 	@LaunchesProjectEvent(UserGroupRemovedProjectEvent.class)
 	@PreAuthorize("hasRole('ROLE_ADMIN') or hasPermission(#project, 'canManageLocalProjectSettings')")
 	public void removeUserGroupFromProject(Project project, UserGroup userGroup) throws ProjectWithoutOwnerException {
+		Set<UserGroupJoin> userGroupJoins = userGroup.getUsers();
```

Comment:
Instead of calling `getUsers` and making the collection eager loading in the UserGroup can you change this to call `userGroupJoinRepository.findUsersInGroup(userGroup);`

Reply:
Please see aa963b11bf9e67e6e18f08d357096ded36438945 and